client/android/WorldScope/app/src/main/java/fragment/ViewVideoControlFragment.java

```diff
@@ -0,0 +1,200 @@
+    }
+
+    // Initialize and run hide controls every 5 secs
+    private void setFadeButtonTimerTask() {
+        timer = new Timer();
+        timerTask = new TimerTask() {
+            public void run() {
+                hideControls();
+            }
+        };
+
+        // Starts hiding controls every 5000ms ~ 5s
+        timer.schedule(timerTask, 5000, 5000);
+    }
+
+    public void toggleControlVisibility() {
+        if(isShown) {
+            // Hide control
+            hideControls();
+        } else {
+            // Show control and reset the timer
+            restartHideButtonTimerTask();
+        }
+    }
+
+    // Interrupt and restart the hide button task
+    private void restartHideButtonTimerTask() {
+        showControls();
+        // Touch event may occur before initialization complete
+        if(timer != null) {
+            timer.cancel();
+        }
+        setFadeButtonTimerTask();
+    }
+
+    // Show the control
+    private void showControls() {
+        isShown = true;
+        getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                view.animate()
+                        .alpha(1.0f)
+                        .setDuration(150)
+                        .setListener(new AnimatorListenerAdapter() {
+                            @Override
+                            public void onAnimationEnd(Animator animation) {
+                                super.onAnimationEnd(animation);
+                                view.setVisibility(View.VISIBLE);
+                            }
+                        });
+            }
+        });
+    }
+
+    // Hide the control
+    private void hideControls() {
+        isShown = false;
+        getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                view.animate()
+                        .alpha(0.0f)
+                        .setDuration(150)
+                        .setListener(new AnimatorListenerAdapter() {
+                            @Override
+                            public void onAnimationEnd(Animator animation) {
+                                super.onAnimationEnd(animation);
+                                view.setVisibility(View.GONE);
+                            }
+                        });
+            }
+        });
+    }
+
+    /**
+     * This interface must be implemented by activities that contain this
+     * fragment to allow an interaction in this fragment to be communicated
+     * to the activity and potentially other fragments contained in that
+     * activity.
+     * <p/>
+     * See the Android Training lesson <a href=
+     * "http://developer.android.com/training/basics/fragments/communicating.html"
+     * >Communicating with Other Fragments</a> for more information.
+     */
+    public interface OnViewVideoControlFragmentListener {
+        // TODO: Update argument type and name
```

Comment:
Its part of refactoring, not really suitable but is a followup to the previous pr